server/src/main/java/org/opensearch/common/cache/stats/CacheStatsHolder.java

```diff
@@ -154,10 +161,17 @@ private boolean internalIncrementHelper(
     }
 
     /**
-     * Produce an immutable version of these stats.
+     * Produce an immutable version of these stats, aggregated according to levels.
+     * If levels is null, do not aggregate and return an immutable version of the original tree.
      */
-    public ImmutableCacheStatsHolder getImmutableCacheStatsHolder() {
-        return new ImmutableCacheStatsHolder(statsRoot.snapshot(), dimensionNames);
+    public ImmutableCacheStatsHolder getImmutableCacheStatsHolder(String[] levels) {
+        List<String> levelsList;
+        if (levels == null) {
+            levelsList = dimensionNames;
+        } else {
+            levelsList = Arrays.asList(levels);
+        }
+        return new ImmutableCacheStatsHolder(this.statsRoot, levelsList, dimensionNames, storeName);
```

Comment:
This computation can be deferred to filterLevels method within ImmutableCacheStatsHolder since dimensionNames is anyway being passed